### DISCUSSION TITLE
Possible wrong response serializing with Pydantic model

### DISCUSSION THREAD

**Discussion by Gipssik** [fastapi/fastapi#6303] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
if isinstance(res, BaseModel):
    read_with_orm_mode = getattr(res.__config__, "read_with_orm_mode", None)
    if read_with_orm_mode:
        # Let from_orm extract the data from this model instead of converting
        # it now to a dict.
        # Otherwise there's no way to extract lazy data that requires attribute
        # access instead of dict iteration, e.g. lazy relationships.
        return res
```

### Description

I am using FastAPI Users with Beanie ODM.
I noticed that response serialized by FastAPI contains `id` field with `null` value.
I started debugging my code and noticed that FastAPI checks if `orm_mode` is turned on in a strange way: [code](https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L69).
It always returns `None`, because Pydantic model's config doesn't contain this field even if `orm_mode = True`.
Serialization works correct only if you serialize object with Pydantic manually like this:
```
return [UserRead.from_orm(user) for user in users]
```
I think this is a bug and it shouldn't work in this way.

### Operating System

Linux

### FastAPI Version

0.75.2

### Python Version

3.10.5

**Comment by JarroVGIT** [comment, marked as answer]
Isn't that exactly as is intended? The `read` in `read_with_orm_mode` should be considered past tense (as in `was_this_object_read_with_orm_mode`). I think your Beanie stuff may add an 'id' field?

**Comment by Gipssik** [comment]
@JarroVGIT Oh, I see, you're absolutely right.
